MiniGamesBox Classic/src/main/java/plugily/projects/minigamesbox/classic/kits/KitRegistry.java

```diff
@@ -56,15 +59,68 @@ public void registerKit(Kit kit) {
       plugin.getDebugger().debug("Kit " + kit.getKey() + " can't be added as its already registered");
       return;
     }
-    FileConfiguration config = ConfigUtils.getConfig(plugin, "kits");
-    if(!config.getBoolean("Enabled-Game-Kits." + kit.getKey(), false)) {
+
+    initializeKitConfig(kit);
+    kit.saveKitsConfig();
+    plugin.getDebugger().debug("Kit " + kit.getKey() + "'s config files have been initialized");
+
+    if(!configurationSection.getBoolean("enabled", false)) {
       plugin.getDebugger().debug("Kit " + kit.getKey() + " is disabled by kits.yml");
       return;
     }
+
+    loadKitConfig(kit);
+    plugin.getDebugger().debug("Kit " + kit.getKey() + "'s config files have been loaded");
+
     plugin.getDebugger().debug("Registered {0} kit", kit.getKey());
     kits.add(kit);
   }
 
+  public void initializeKitConfig(Kit kit) {
+    ConfigurationSection configurationSection = kit.getKitsConfig().getConfigurationSection(kit.getKey());
+    if (configurationSection == null) {
+      configurationSection = kit.getKitsConfig().createSection(kit.getKey());
+    }
+    configurationSection.set("name", kit.getName());
+    if (!configurationSection.contains("enabled")) {
+      configurationSection.set("enabled", true);
+    }
+    int currentItem = 0;
+    if (!configurationSection.contains("items")) {
```

Comment:
contains category "inventory"

should look a bit like:
inventory:
  items: 
     SLOTkey: 
          xxx
  armour:
      head: 
      chest: xx

Reply:
The configuration file for kits.yml should be auto generated. After looking at what you have proposed, I'm thinking of doing the config somewhat like this:

BridgeKit:
  Inventory:
    1:
      item:
        name: Diamond sword
        material: DIAMOND_SWORD
      slot: 0
  Armour:
    Helmet:
      name: Diamond helmet
      material: DIAMOND_HELMET

Reply:
Resolved.